### ISSUE TITLE
Shareable link for Reaction Streamers

### ISSUE THREAD

**Issue by npetrangelo** [Chocobozzz/PeerTube#5092]
### Describe the problem to be solved

The problem is rather complex but is summarized well in [this YouTube video](https://www.youtube.com/watch?v=jKFruXvzsGc), along with Zoe Bee's idea for a solution. However, I have a different solution that I believe PeerTube could implement as well as YouTube, and I believe PeerTube would gain an edge by implementing it first. It would also enable streamers with big platforms to react to PeerTube content in a way that actually directs their audience to PeerTube to watch it.

Zoe Bee's criteria were as follows:
1. Provides views to both the streamer and creator
2. Simple as possible for everyone involved (streamer and audience)
3. Usable on most if not all devices.

I believe my solution satisfies all of these criteria.

### Describe the solution you would like

A streamer who is about to react to a PeerTube video would create a special sharable link, that they then share with their audience. When a stream viewer clicks this link to watch the video, the video's playback is controlled by the streamer; pausing when the streamer pauses it, playing when the streamer plays it, and generally staying synchronized with the streamer's view of the video. Simultaneously, the link also processes everyone's views separately, viewers can subscribe to the creator's channel if they so choose, and they can even leave a comment. Finally, when the streamer finishes watching the video, the link functions as a standard sharable link. All of this happens while the viewers are still watching the stream as per usual.

Under the hood, PeerTube would know the account of the viewer if they are logged in, which device they are watching from, and that the streamer generated it for this purpose. This, in principle, is everything PeerTube needs to know to make this work, although it may take some additional initial setup on the streamer's part. On the Twitch mobile app at least, when a stream viewer clicks a link, the stream automatically minimizes into the corner of the screen, so this is not a barrier to this method being usable on mobile.

In conclusion:
- [x] Provides views to both the streamer and creator
- [x] Simple as generating, sharing, and clicking a link
- [x] Usable on most if not all devices
